core/src/test/scala/unit/kafka/controller/KafkaControllerTest.scala

```diff
@@ -51,19 +54,20 @@ class KafkaControllerTest {
         initialBrokerEpoch = 0,
         tokenManager = mock(classOf[DelegationTokenManager]),
         brokerFeatures = mock(classOf[BrokerFeatures]),
-        featureCache = mock(classOf[ZkMetadataCache])
+        featureCache = mock(classOf[ZkMetadataCache]),
+        metricTags = metricTags
       )
 
       // shutdown kafkaController so that metrics are removed
       kafkaController.shutdown()
 
       val mockMetricsGroup = mockMetricsGroupCtor.constructed.get(0)
       val numMetricsRegistered = KafkaController.MetricNames.size
-      verify(mockMetricsGroup, times(numMetricsRegistered)).newGauge(anyString(), any())
-      KafkaController.MetricNames.foreach(metricName => verify(mockMetricsGroup).newGauge(ArgumentMatchers.eq(metricName), any()))
+      verify(mockMetricsGroup, times(numMetricsRegistered)).newGauge(anyString(), any(), ArgumentMatchers.eq(metricTags.asJava))
+      KafkaController.MetricNames.foreach(metricName => verify(mockMetricsGroup).newGauge(ArgumentMatchers.eq(metricName), any(), ArgumentMatchers.eq(metricTags.asJava)))
       // verify that each metric is removed
-      verify(mockMetricsGroup, times(numMetricsRegistered)).removeMetric(anyString())
-      KafkaController.MetricNames.foreach(verify(mockMetricsGroup).removeMetric(_))
+      verify(mockMetricsGroup, times(numMetricsRegistered)).removeMetric(anyString(), ArgumentMatchers.eq(metricTags.asJava))
+      KafkaController.MetricNames.foreach { name => verify(mockMetricsGroup).removeMetric(ArgumentMatchers.eq(name), ArgumentMatchers.eq(metricTags.asJava)) }
```

Comment:
Noting the same here, these changes simply deal with the implications of introducing metric tags to the KafkaController.